Move requirements to a folder

andrey-canon opened the pull request:
## Description
Move requirements into a folder
update settings
add upgrade command

morenol commented on eox_core/settings/test.py:
```diff
@@ -58,3 +60,27 @@ def plugin_settings(settings):  # pylint: disable=function-redefined
 USERNAME_REGEX_PARTIAL = r'[\w .@_+-]+'
 USERNAME_PATTERN = r'(?P<username>{regex})'.format(regex=USERNAME_REGEX_PARTIAL)
+
+PROCTORING_SETTINGS = {
```
Why is this added in this PR?

andrey-canon replied:
the version 1.5.7 tries to import  signals and signals.py tries to import the constants https://github.com/edx/edx-proctoring/blob/1.5.7/edx_proctoring/apps.py#L117,  the previous version was 1.4.0, but that version is not the platform version.

morenol commented on requirements/base.in:
```diff
@@ -0,0 +1,14 @@
+django-waffle
+edx-proctoring
+edx-opaque-keys
+factory-boy
```
Is this a test package?

morenol commented on requirements/base.in:
```diff
@@ -0,0 +1,14 @@
+django-oauth2-provider
+django-waffle
+edx-proctoring
+edx-opaque-keys
```
setup.py has some of these dependencies, should we remove them from there? or change setup.py to use this as dependencies?

In case of the second, I think that we need to change this with edx-opaque-keys[django] in order to prevent error when this package is installed using `pip install eox-core`

andrey-canon commented:
> I added a few comments, please add an entry in the changelog.

@morenol  does it make any sense to  add refactor or quality changes  in the changelog ?, I mean, this pr doesn't modify, add or fix anything, so if someone fixes a typo, a new entry will be required?, imo the changelog only should contain functionality changes, btw I have already added  the entry.

morenol commented on Makefile:
```diff
@@ -35,3 +35,8 @@ python-quality-test:
 
 javascript-quality-test:
 	./node_modules/.bin/eslint ./eox_core/**/*.js*
+
```
```suggestion

upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
```
In order to get the generated .txt files have this instead of the `pip-compile` complete command.

andrey-canon replied:
Updated to the last version on the cookiecutter

morenol commented on Makefile:
```diff
@@ -35,3 +35,8 @@ python-quality-test:
 	./node_modules/.bin/eslint ./eox_core/**/*.js*
+
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	pip install -q pip-tools
```
I think that we should create a pip-tools.in and a pip-tools.txt files with this package

morenol commented on Makefile:
```diff
@@ -35,3 +35,8 @@ python-quality-test:
 javascript-quality-test:
 	./node_modules/.bin/eslint ./eox_core/**/*.js*
+
+upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
```
I think that we should roon pip-compile with this flag:  `--upgrade`

morenol commented:
> @morenol does it make any sense to add refactor or quality changes in the changelog ?, I mean, this pr doesn't modify, add or fix anything, so if someone fixes a typo, a new entry will be required?, .

You are right, but maybe the change related to the PROCTORING_SETTINGS is worth

andrey-canon commented:
> ht, but maybe the change related to the PROCTORING_SETTINGS is worth

@morenol  something like `Add proctoring test settings since this had the wrong proctoring version` ???

Squirrel18 commented on requirements/base.in:
```diff
@@ -0,0 +1,12 @@
+django-oauth2-provider
+django-waffle
+edx-proctoring
+edx-opaque-keys[django]
```
Where's the six package? I think we should explicitly add it. @andrey-canon
